tests/test_nhg.py

```diff
@@ -329,12 +721,35 @@ def asic_route_nhg_fvs(k):
             route_count += 1
 
         # Wait and check ASIC DB the count of nexthop groups used
-        asic_db.wait_for_n_keys("ASIC_STATE:SAI_OBJECT_TYPE_NEXT_HOP_GROUP", MAX_ECMP_COUNT)
+        asic_db.wait_for_n_keys(self.ASIC_NHG_STR, MAX_ECMP_COUNT)
 
         # Wait and check ASIC DB the count of routes
-        asic_db.wait_for_n_keys("ASIC_STATE:SAI_OBJECT_TYPE_ROUTE_ENTRY", asic_routes_count + MAX_ECMP_COUNT)
+        asic_db.wait_for_n_keys(self.ASIC_RT_STR, asic_routes_count + MAX_ECMP_COUNT)
         asic_routes_count += MAX_ECMP_COUNT
 
+        # Add a route with labeled NHs
+        asic_nhs_count = len(asic_db.get_keys(self.ASIC_NHS_STR))
+        route_ipprefix = gen_ipprefix(route_count)
+        fvs = swsscommon.FieldValuePairs([('nexthop', '10.0.0.1,10.0.0.3'),
+                                          ('mpls_nh', 'push1,push3'),
+                                          ('ifname', 'Ethernet0,Ethernet4')])
+        ps.set(route_ipprefix, fvs)
+        route_count += 1
+
+        # A temporary route should be created
+        asic_db.wait_for_n_keys(self.ASIC_RT_STR, asic_routes_count + 1)
+
+        # A NH should be elected as the temporary NHG and it should be created
+        # as it doesn't exist.
+        asic_db.wait_for_n_keys(self.ASIC_NHS_STR, asic_nhs_count + 1)
+
+        # Delete the route.  The route and the added labeled NH should be
+        # removed.
+        ps._del(route_ipprefix)
+        route_count -= 1
+        asic_db.wait_for_n_keys(self.ASIC_RT_STR, asic_routes_count)
+        asic_db.wait_for_n_keys(self.ASIC_NHS_STR, asic_nhs_count)
+
```

Comment:
How about moving this to new test case to test the nexthop group resource exhaustion logic in 'nhgOrch'. It's good to have it's own test case so failures will be easier to debug. You can set the nexthop group resource limit by adding this key in state DB: SWITCH_CAPABILITY|switch

Reply:
Ignore this, I see you already added one below.